### ISSUE TITLE
`flux_bootstrap_git` fails with `components`

### ISSUE THREAD

**Issue by ppodevlab** [fluxcd/terraform-provider-flux#378]
# Description
Hi, 
we are testing the terraform provider in version 0.23.0 with the new resource `flux_bootstrap_git` and we are facing some issues.

## Configuration 

###variables.tf
```
k8s_apps_flux_version = "v0.39.0"
k8s_apps_flux_git_repository_ssh_url = "ssh://git@mybitbucket.org/project/repo.git"
k8s_apps_flux_git_branch = "main"
k8s_apps_flux_components = ["source-controller", "kustomize-controller", "helm-controller", "notification-controller"]
k8s_apps_flux_components_extra = ["image-reflector-controller", "image-automation-controller"]
```

### main.tf
```
resource "flux_bootstrap_git" "flux" {
  url     = var.k8s_apps_flux_git_repository_ssh_url
  path    = "clusters/test"
  branch  = var.k8s_apps_flux_git_branch
  version = var.k8s_apps_flux_version
  components = toset(var.k8s_apps_flux_components)
  components_extra = toset(var.k8s_apps_flux_components_extra)
  ssh = {
    username    = "git"
    private_key = var.private_key
  }
}
```
When executing the script i got
```
│ Path:
│ Error: target must be a pointer, got []string, which is a slice
```

Note: For extra components it works, so the issue must be on the components

**Comment by phillebaba** [comment]
This has been fixed in #371 which was released in v0.24.0. Could your try upgrading the provider and try again?

**Comment by ppodevlab** [comment]
> This has been fixed in #371 which was released in v0.24.0. Could your try upgrading the provider and try again?

confirmed! thanks!
